registry/config.go

```diff
@@ -23,12 +23,46 @@ type ServiceOptions struct {
 	// V2Only controls access to legacy registries.  If it is set to true via the
 	// command line flag the daemon will not attempt to contact v1 legacy registries
 	V2Only bool `json:"disable-legacy-registry,omitempty"`
+
+	// Registries holds information associated with the specified registries.
+	Registries []Registry `json:"registries,omitempty"`
 }
 
 // serviceConfig holds daemon configuration for the registry service.
 type serviceConfig struct {
 	registrytypes.ServiceConfig
-	V2Only bool
+	V2Only     bool
+	Registries map[string]Registry
+}
+
+// Registry holds information for a registry and its mirrors.
+type Registry struct {
+	// Prefix is used for the lookup of endpoints, where the given registry
+	// is selected when its Prefix is a prefix of the passed reference, for
+	// instance, Prefix:"docker.io/opensuse" will match a `docker pull
+	// opensuse:tumleweed`.
+	Prefix string `json:"Prefix,omitempty"`
+	// The mirrors will be selected prior to the registry during lookup of
+	// endpoints.
+	Mirrors []Mirror `json:"Mirrors,omitempty"`
+	// Host is the host of the registry (e.g., "docker.io")
+	Host string
+	// This avoids to parse the URL of each new endpoint lookup.
+	URL *url.URL
+	// True if the host is "docker.io".
+	Official bool
+	// True if URL.Scheme == "https".
```

Comment:
Also, this should not be part of the config.

Reply:
It's not JSON configurable, so I assumed it's okay to have it. The reasoning behind is to avoid parsing the URL for each lookup or avoiding redundant comparisons, but I can work around that.

Reply:
Again, this might get picked up.

I think you can add `json:"-"` as the struct tag for the fields that should be ignored.